### PR TITLE
Add support for WorkGroup CRD

### PR DESCRIPTION
**Pull request by TiberiuGC**
Issue #, if available:

Description of changes:
- adding support for WorkGroup CRD
- adding relevant integration tests

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: apis/v1alpha1/work_group.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	// and whether workgroup's settings (specified with EnforceWorkGroupConfiguration)
+	// in the WorkGroupConfiguration override client-side settings. See WorkGroupConfiguration$EnforceWorkGroupConfiguration.
+	Configuration *WorkGroupConfiguration `json:"configuration,omitempty"`
+	// The workgroup description.
+	Description *string `json:"description,omitempty"`
+	// The workgroup name.
+	// +kubebuilder:validation:Required
+	Name *string `json:"name"`
+	// A list of comma separated tags to add to the workgroup that is created.
+	Tags []*Tag `json:"tags,omitempty"`
```
Comment (a-hilaly) [inline apis/v1alpha1/work_group.go:51]: Can we also support tag updates?

File: templates/hooks/workgroup/sdk_read_one_post_request.go.tpl [hunk 2]
```diff
@@ -0,0 +1,7 @@
+	// Athena API returns a 400 instead of a 404 when a workgroup is not found.
+	if err != nil {
+		if reqErr, ok := ackerr.AWSRequestFailure(err); ok && reqErr.StatusCode() == 400 && 
+			strings.Contains(reqErr.Error(), "is not found") {
+			return nil, ackerr.NotFound
+		}
+	}
```
Comment (a-hilaly) [inline templates/hooks/workgroup/sdk_read_one_post_request.go.tpl:7]: We have a code generator option that helps remapping the code errors https://aws-controllers-k8s.github.io/community/docs/contributor-docs/code-generator-config/#exceptions-correcting-exception-codes

File: apis/v1alpha1/work_group.go (hunk 1 above)
Comment (a-hilaly) [reply apis/v1alpha1/work_group.go:51]: Can we have it in this PR @TiberiuGC or are planning on doing it seperately?

Comment (michaelhtm) [reply apis/v1alpha1/work_group.go:51]: is this done?

**Review by a-hilaly — APPROVED** [review]
This is great, thank you @TiberiuGC !
/lgtm

**Pull request merged**